keras_nlp/models/gpt_neox/rotary_embedding.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+from tensorflow import keras
+
+class RotaryEmbedding(keras.layers.Layer):
+    def __init__(self, dim, rotary_emb_base=10000):
+        super().__init__()
+        self.dim = int(dim)
+        self.rotary_emb_base = rotary_emb_base
+
+    def build(self, input_shape):
+        super().build(input_shape)
+        self.inverse_freq = self.add_weight(
+            "inverse_freq", shape=(self.dim // 2,), dtype=tf.float32
+        )
+        range = tf.range(start=0, limit=self.dim, delta=2, dtype="float32")
+
+        self.inverse_freq.assign(
+            1.0 / (self.rotary_emb_base ** (range / self.dim))
+        )
+
+    @staticmethod
```

Comment:
why bother marking this a static method and leaving it public?

seems like we could just leave this as a private `_apply_rotary_pos_emb` regular method for now (and let the fact that this does not access anything on self be incidental)

Reply:
sounds good

Reply:
this is still marked as a staticmethod, I think there is no need.